tests/test_operator.py

```diff
@@ -947,6 +947,43 @@ def test_function_dataobj(self, so, to, pad, expected):
 
         assert u_arg_shape == expected
 
+    def test_illegal_override(self):
+        grid0 = Grid(shape=(11, 11))
+        grid1 = Grid(shape=(13, 13))
+
+        a0 = Function(name='a', grid=grid0)
+        b0 = Function(name='b', grid=grid0)
+        a1 = Function(name='a', grid=grid1)
+        b1 = Function(name='b', grid=grid1)
+
+        op = Operator(Eq(a0, a0 + b0 + 1))
+        op.apply()
+
+        try:
+            op.apply(a=a1, b=b1)
```

Comment:
running it locally, this runs fine, doesn't end up in the except section

Reply:
as expected, otherwise CI would be red ? not sure I follow?

Reply:
>Check it's hitting the right error msg

This is never actually checked as op.apply runs fine

Reply:
ah, I see. Good spot. DO you know why? Otherwise I'll take a look

Reply:
Both have same grid so still fine, `op.apply(a=a1, b=b0)` should do the trick

Reply:
ah yes! can you patch it?

Reply:
or I will tmr morning no probs